inc/integrations/class-coral.php

```diff
@@ -193,4 +295,157 @@ public function build_jwt( array $credentials ): string {
 	public function base64url_encode( string $data ): string {
 		return rtrim( strtr( base64_encode( $data ), '+/', '-_' ), '=' );
 	}
+
+
+	/**
+	 * Retrieve a username record. These functions use the post_title column
+	 * for the email address and the post_excerpt column for the username.
+	 *
+	 * @param string $email    The email address of the user.
+	 * @return string The username, or a blank string if none is set.
+	 */
+	private function get_username( $email ) : string {
+		global $wpdb;
+
+		$username = $wpdb->get_var( $wpdb->prepare( 
+			"SELECT post_excerpt 
+				FROM {$wpdb->posts} 
+			WHERE 
+				post_title=%s AND
+				post_type=%s AND
+				post_status='publish'
+			LIMIT 1 ",
+			[
+				$email,
+				$this->post_type,
+			]
+		) );
+
+		return $username ?? '';
+	}
+
+	/**
+	 * Retrieve a username record's post ID. These functions use the post_title column
+	 * for the email address and the post_excerpt column for the username.
+	 *
+	 * @param string $email    The email address of the user.
+	 * @return int The post ID, or 0 if none is found.
+	 */
+	private function get_username_post_id( $email ) : int {
```

Comment:
I'm not sure we should be using the email address as the relevant information to match with a Pico user, since you can update your email address in Pico. Instead, could we store the Pico user ID and use that to lookup usernames?